plugin/src/main/scala/chisel3/internal/plugin/ChiselComponent.scala

```diff
@@ -82,7 +82,12 @@ class ChiselComponent(val global: Global, arguments: ChiselPluginArguments)
     // Checking for all chisel3.internal.NamedComponents, but since it is internal, we instead have
     // to match the public subtypes
     private val shouldMatchNamedComp: Type => Boolean =
-      shouldMatchGen(tq"chisel3.Data", tq"chisel3.MemBase[_]", tq"chisel3.VerificationStatement")
+      shouldMatchGen(
+        tq"chisel3.Data",
+        tq"chisel3.MemBase[_]",
+        tq"chisel3.VerificationStatement",
+        tq"chisel3.experimental.HasChiselName"
+      )
```

Comment:
I need to include some lines from below (lines 182-190) to mention what I don't love about this current implementation:
```scala
        else if (shouldMatchNamedComp(tpe)) {
          val str = stringFromTermName(name)
          // Starting with '_' signifies a temporary, we ignore it for prefixing because we don't
          // want double "__" in names when the user is just specifying a temporary
          val prefix = if (str.head == '_') str.tail else str
          val newRHS = transform(rhs)
          val prefixed = q"chisel3.experimental.prefix.apply[$tpt](name=$prefix)(f=$newRHS)"
          val named = q"chisel3.internal.plugin.autoNameRecursively($str)($prefixed)"
          treeCopy.ValDef(dd, mods, name, tpt, localTyper.typed(named))
```

So as we can see, things that extend `HasChiselName` will now be included as matching in that `if` block. This means they will be wrapped in `autoNameRecursively`, yet they are not of type `HasId` so they aren't _actually_ namable (that `autoNameRecursively` is effectively a no-op).

Technically this has no impact on the output, but it does
1. Slightly slow down compilation
2. Slightly increase the size of the compiled user code
3. Slightly slow down elaboration time
4. Show up in stack traces which may make it appear to be doing something, but it doesn't.

I think we should not generate the `autoNameRecursively` for things that extend `HasChiselName` (but are not also `Data` obviously).